emark/views.py

```diff
@@ -93,3 +97,67 @@ def get(self, request, *args, **kwargs):
                 "Cache-Control": "no-cache, no-store, must-revalidate",
             },
         )
+
+
+class DashboardView(LoginRequiredMixin, UserPassesTestMixin, TemplateView):
+    """Show a dashboard of available email classes."""
+
+    template_name = "emark/dashboard.html"
+
+    def test_func(self):
+        return self.request.user.is_staff
+
+    def get_emails(self):
+        hidden_classes = [
+            message.MarkdownEmail.__name__,
+            *conf.get_settings().DASHBOARD_HIDDEN_CLASSES,
+        ]
+        emails = [
+            {
+                "app_label": email_class.__module__.split(".")[0],
+                "class_name": email_class.__name__,
+                "doc": email_class.__doc__ or "",
+                "detail_url": reverse(
+                    "emark:email-preview", args=[email_class.__name__]
+                ),
+            }
+            for email_class in utils.get_subclasses(message.MarkdownEmail)
+            if email_class.__name__ not in hidden_classes
+        ]
+        return sorted(
+            emails, key=lambda email: (email["app_label"], email["class_name"])
+        )
+
+    def get_context_data(self, **kwargs):
+        return super().get_context_data(**kwargs) | {
+            "emails": self.get_emails(),
+        }
+
+
+class EmailPreviewView(LoginRequiredMixin, UserPassesTestMixin, TemplateView):
+    """Render a preview of the email."""
+
+    template_name = "emark/preview.html"
+
+    def test_func(self):
+        return self.request.user.is_staff
+
+    def dispatch(self, request, *args, **kwargs):
+        self.email_class = self.get_email_class(kwargs["email_class"])
+        if not self.email_class:
+            raise Http404()
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_email_class(self, email_class):
+        for cl in utils.get_subclasses(message.MarkdownEmail):
+            if cl.__name__ == email_class:
```

Comment:
If you add a registry you can include a unique classifier. Loading classes based on a user input, even if limited by subclasses, is a slippery slope.